pkg/common/client.go

```diff
@@ -125,6 +127,31 @@ func newTLSConfig(clientCert, clientKey, caCert string) (*tls.Config, error) {
 	return config, nil
 }
 
+func MakeAdminClient(clientID string, kafkaAuthCfg *utils.KafkaAuthConfig, bootstrapServers []string) (sarama.ClusterAdmin, error) {
+	saramaConf := sarama.NewConfig()
+	saramaConf.Version = sarama.V1_1_0_0
+	saramaConf.ClientID = clientID
+
+	if kafkaAuthCfg != nil {
+		// tls
+		if kafkaAuthCfg.TLS != nil {
+			saramaConf.Net.TLS.Enable = true
+			tlsConfig, err := NewTLSConfig(kafkaAuthCfg.TLS.Usercert, kafkaAuthCfg.TLS.Userkey, kafkaAuthCfg.TLS.Cacert)
+			if err != nil {
+				return nil, err
+			}
+			saramaConf.Net.TLS.Config = tlsConfig
+		}
+		// SASL
+		if kafkaAuthCfg.SASL != nil {
+			saramaConf.Net.SASL.Enable = true
+			saramaConf.Net.SASL.User = kafkaAuthCfg.SASL.User
+			saramaConf.Net.SASL.Password = kafkaAuthCfg.SASL.Password
+		}
+	}
+	return sarama.NewClusterAdmin(bootstrapServers, saramaConf)
+}
+
```

Comment:
Should configuration of the kafka auth (`saramaConf`) be in it's own utility function instead of the directly in the logic for creating an ClusterAdmin?   The same logic will be needed for creating Producers & Consumers right?  This is why we pass the sarama.Config into the `CreateAdminClient()` in the distributed implementation (see distributed/common/kafka/admin/admin.go).

Reply:
I'd like to give the entire work on saramaConf and "client" some re-thoughts:

* https://github.com/knative-sandbox/eventing-kafka/issues/188